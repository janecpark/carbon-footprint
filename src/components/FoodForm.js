import React, { useState, useContext } from 'react';
import { useHistory } from 'react-router-dom';
import FormContext from '../FormContext';
import ResultContext from '../ResultContext';
import CarbonApi from '../CarbonApi';
import Slider from 'react-rangeslider';
import './FoodForm.css';
import 'react-rangeslider/lib/index.css';

const FoodForm = () => {
  const history = useHistory();
  const { formData, setFormData } = useContext(FormContext);
  const { result, setResult } = useContext(ResultContext);
  const [food, setFood] = useState(0);
  const handleChange = (value) => {
    setFood(value);
  };
  async function handleSubmit(e) {
    e.preventDefault();
    setFormData((prevState) => ({
      ...prevState,
      input_footprint_shopping_food_meatfisheggs_default: food,
    }));
    console.log(formData);
    let res = await CarbonApi.travelData(formData);
    let parsedResult = JSON.parse(res.result);
    setResult(parsedResult);
    console.log(parsedResult);
    history.push('/result');
  }
  const labels = {
    0: '0',
    362: 'Average',
    724: ' 2x',
    1086: '3x',
  };
  // const style = {
  //   backgroundImage: `url(${chicken})`,
  //   height: '100vh',
  //   backgroundSize: 'cover',
  //   width: '100vw',
  // };
  return (
    <form className="slider custom-labels container" onSubmit={handleSubmit}>
      <h4> How much meat, fish, eggs does the average person in your household eat?</h4>
      <Slider
        orientation="horizontal"
        onChange={handleChange}
        min={0}
        max={1086}
        tooltip={false}
        value={food}
        labels={labels}
      />
      <button className='btn btn-success mt-3'>Get results</button>
    </form>
  );
};

export default FoodForm;
