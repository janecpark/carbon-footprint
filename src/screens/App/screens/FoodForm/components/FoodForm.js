import React, { useState, useContext } from 'react';
import { useHistory } from 'react-router-dom';
import FormContext from '../../../FormContext';
import ResultContext from '../../../ResultContext';
import CarbonApi from '../../../CarbonApi';
import Slider from 'react-rangeslider';
import 'react-rangeslider/lib/index.css';
import getTotal from './CalHelper';
import ActionContext from '../../../ActionContext';

const FoodForm = () => {
  const history = useHistory();
  const { formData, setFormData } = useContext(FormContext);
  const { setResult } = useContext(ResultContext);
  const { setAction } = useContext(ActionContext);
  const [food, setFood] = useState(0);
  const handleChange = (value) => {
    setFood(value);
  };
  async function handleSubmit(e) {
    e.preventDefault();
    setFormData((prevState) => ({
      ...prevState,
      input_footprint_shopping_food_meatfisheggs_default: food || 1086,
    }));
    try {
      let res = await CarbonApi.travelData(formData);
      let data = JSON.parse(res.result);
      if (data.response.hasOwnProperty('failed')) {
        setResult((e) => ({ ...e, errors: data.response.error_message}));
      }else{
        setResult({
          food: Number(parseFloat(data.response.result_food_total).toFixed(2)),
          housing: Number(
            parseFloat(data.response.result_housing_total).toFixed(2)
          ),
          transport: Number(
            parseFloat(data.response.result_transport_total).toFixed(2)
          ),
          total: getTotal(
            data.response.result_food_total,
            data.response.result_housing_total,
            data.response.result_transport_total
          ),
        });
        let data_result = JSON.parse(data.response.result_takeaction_pounds);
        setAction({
          public_transport: data_result.take_public_transportation,
          carpool: data_result.carpool_to_work,
          low_carbon_diet: data_result.low_carbon_diet,
          organic: data_result.go_organic,
          fridge: data_result.energy_star_fridge,
          green_electricity: data_result.purchase_green_electricity,
        });
      }
    } catch (err) {
      setResult((e) => ({ ...e, errors: { ...err } }));
    }
    history.push('/result');
  }
  const labels = {
    0: '0',
    362: 'Average',
    724: ' 2x',
    1086: '3x',
  };

  return (
    <div>
      <form className="slider custom-labels container" onSubmit={handleSubmit}>
        <h4>
          How much meat, fish, eggs does the average person in your household
          eat?
        </h4>
        <Slider
          orientation="horizontal"
          onChange={handleChange}
          min={0}
          max={1086}
          tooltip={false}
          value={food}
          labels={labels}
        />
        <button className="btn btn-success mt-3">Get results</button>
      </form>
    </div>
  );
};

export default FoodForm;
