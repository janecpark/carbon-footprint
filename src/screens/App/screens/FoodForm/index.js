import FoodForm from './components/FoodForm';

export default FoodForm;