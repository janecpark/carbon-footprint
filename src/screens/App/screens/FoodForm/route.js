import { routeConstants } from '../../../../shared/constants';
import FoodForm from './';

export default {
  path: routeConstants.FOOD.route,
  component: FoodForm,
};
