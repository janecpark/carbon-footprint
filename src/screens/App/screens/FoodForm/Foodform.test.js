import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import FoodForm from './components/FoodForm';
import {
  UserProvider,
  FormProvider,
  ResultProvider,
  ActionProvider,
} from '../testUtils';

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <FoodForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
});

it('matches snapshot', function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <FoodForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders text on page', () => {
  const { getByText } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <FoodForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  const element = getByText(/Get results/);
  expect(element).toBeInTheDocument();
});

test('Full app navigating when button is clicked', () =>{
  const { getByText } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <FoodForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  const btn = getByText('Get results')
  fireEvent.click(btn)
  expect(screen.getByText(/No results/)).toBeInTheDocument()
})