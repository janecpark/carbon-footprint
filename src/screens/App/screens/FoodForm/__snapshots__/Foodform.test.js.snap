// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form
      class="slider custom-labels container"
    >
      <h4>
        How much meat, fish, eggs does the average person in your household eat?
      </h4>
      <div
        aria-orientation="horizontal"
        aria-valuemax="1086"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rangeslider rangeslider-horizontal"
      >
        <div
          class="rangeslider__fill"
          style="width: 0px;"
        />
        <div
          class="rangeslider__handle"
          style="left: 0px;"
          tabindex="0"
        >
          <div
            class="rangeslider__handle-label"
          />
        </div>
        <ul
          class="rangeslider__labels"
        >
          <li
            class="rangeslider__label-item"
            data-value="1086"
            style="left: 0px;"
          >
            3x
          </li>
          <li
            class="rangeslider__label-item"
            data-value="724"
            style="left: 0px;"
          >
             2x
          </li>
          <li
            class="rangeslider__label-item"
            data-value="362"
            style="left: 0px;"
          >
            Average
          </li>
          <li
            class="rangeslider__label-item"
            data-value="0"
            style="left: 0px;"
          >
            0
          </li>
        </ul>
      </div>
      <button
        class="btn btn-success mt-3"
      >
        Get results
      </button>
    </form>
  </div>
</DocumentFragment>
`;
